internal/apis/certmanager/v1alpha2/types_certificate.go

```diff
@@ -501,3 +507,41 @@ type CertificateAdditionalOutputFormat struct {
 	// Certificate's target Secret.
 	Type CertificateOutputFormatType `json:"type"`
 }
+
+// NameConstraints is a type to represent x509 NameConstraints
+type NameConstraints struct {
+	// if true then the name constraints are marked critical.
+	//
+	// +optional
+	Critical bool `json:"critical,omitempty"`
```

Comment:
What should the default value be for Critical? @SgtCoDFish

Reply:
I think the default should be false. Users need to configure this to true to make the constraints critical. The behaviour is similar in OpenSSL as in users would need to mention `critical` in name constraints for using this parameter